### PR TITLE
[ASTGen] NFC: Format with swift-format

### PR DESCRIPTION
**Pull request by hamishknight**
This uses the same formatting configuration as sourcekit-lsp.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please smoke test

File: lib/ASTGen/.swift-format [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "version": 1,
+    "lineLength": 120,
```
Comment (AnthonyLatsis) [inline lib/ASTGen/.swift-format:3]: FYI we settled on 160 [here](https://github.com/apple/swift/pull/66044#discussion_r1206085959), which motivated https://github.com/apple/swift-syntax/pull/1714.

How about a symlink to the swift-syntax configuration file for everything inside `lib` instead? Consistent formatting across our Swift repos sounds really nice.

Comment (AnthonyLatsis) [reply lib/ASTGen/.swift-format:3]: > but AFAIK we want to standardize on 120

Oh, that’s new. Totally fine by me! I still incline to a symlink until swift-syntax transitions though. I’m wondering what we can do to standardize the Swift formatting in these repositories without having to maintain multiple config files.

Comment (ahoppen) [reply lib/ASTGen/.swift-format:3]: Yes, we agreed to use 120 columns. We should update swift-syntax to 120 as well at some point but I haven’t gotten to it yet.

> I still incline to a symlink until swift-syntax transitions though

If we were to symlink swift-syntax’s config file to swift, it means that you can’t work on swift-syntax as a standalone package anymore. Similar for sourcekit-lsp. I think the best thing to do, is to have a separate config file in each repo and add a CI check that all the config files are the same.

File: lib/ASTGen/Sources/LLVMJSON/LLVMJSON.swift [hunk 2]
```diff
@@ -653,13 +666,15 @@ extension LLVMJSONEncoding.KeyedContainer: KeyedEncodingContainerProtocol {
...
+  mutating func encode<T>(_ value: T, forKey key: Key) throws where T: Encodable {
     let valuePtr = JSON_object_setNewValue(objectPtr, key.stringValue)
     let encoder = LLVMJSONEncoding(to: valuePtr, codingPath: codingPath + [key])
     try value.encode(to: encoder)
   }
 
-  mutating func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type, forKey key: Key) -> KeyedEncodingContainer<NestedKey> where NestedKey : CodingKey {
+  mutating func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type, forKey key: Key) -> KeyedEncodingContainer<
+    NestedKey
+  > where NestedKey: CodingKey {
```
Comment (ahoppen) [inline lib/ASTGen/Sources/LLVMJSON/LLVMJSON.swift:677]:
```suggestion
  mutating func nestedContainer<NestedKey: CodingKey>(
    keyedBy keyType: NestedKey.Type, 
    forKey key: Key
  ) -> KeyedEncodingContainer<NestedKey> {
```

File: lib/ASTGen/Sources/LLVMJSON/LLVMJSON.swift [hunk 3]
```diff
@@ -511,17 +519,22 @@ extension LLVMJSONDecoding.UnkeyedContainer: UnkeyedDecodingContainer {
...
+      .init(
+        codingPath: codingPath + [IndexKey(intValue: currentIndex)],
+        debugDescription: "type misatch"
+      )
+    )
   }
 
-  mutating func nestedContainer<NestedKey>(keyedBy type: NestedKey.Type) throws -> KeyedDecodingContainer<NestedKey> where NestedKey : CodingKey {
+  mutating func nestedContainer<NestedKey>(keyedBy type: NestedKey.Type) throws -> KeyedDecodingContainer<NestedKey>
+  where NestedKey: CodingKey {
```
Comment (ahoppen) [inline lib/ASTGen/Sources/LLVMJSON/LLVMJSON.swift:537]:
```suggestion
  mutating func nestedContainer<NestedKey: CodingKey>(
    keyedBy type: NestedKey.Type
  ) throws -> KeyedDecodingContainer<NestedKey> {
```

File: lib/ASTGen/Sources/LLVMJSON/LLVMJSON.swift [hunk 4]
```diff
@@ -381,12 +384,14 @@ extension LLVMJSONDecoding.KeyedContainer: KeyedDecodingContainerProtocol {
...
-  func decodeIfPresent<T>(_ type: T.Type, forKey key: Key) throws -> T? where T : Decodable {
+  func decodeIfPresent<T>(_ type: T.Type, forKey key: Key) throws -> T? where T: Decodable {
     guard contains(key) else { return nil }
     return try decode(type, forKey: key)
   }
 
-  func nestedContainer<NestedKey>(keyedBy type: NestedKey.Type, forKey key: Key) throws -> KeyedDecodingContainer<NestedKey> where NestedKey : CodingKey {
+  func nestedContainer<NestedKey>(keyedBy type: NestedKey.Type, forKey key: Key) throws -> KeyedDecodingContainer<
+    NestedKey
+  > where NestedKey: CodingKey {
```
Comment (ahoppen) [inline lib/ASTGen/Sources/LLVMJSON/LLVMJSON.swift:394]:
```suggestion
  func nestedContainer<NestedKey: CodingKey>(
    keyedBy type: NestedKey.Type, 
    forKey key: Key
  ) throws -> KeyedDecodingContainer<NestedKey> {
```

**Review by ahoppen — APPROVED** [review]
I’ve got a few places where I think that swift-format’s default line wrapping behavior can be improved but I’m fine with adjusting them in a follow-up PR if you want to merge this to unblock other PRs.

**Comment by hamishknight** [comment]
Merging so I can base #69326 on main, will do a follow-up for the formatting fixups

**Comment by hamishknight** [comment]
Opened #69336 for the formatting tweaks

File: lib/ASTGen/.swift-format (hunk 1 above)
Comment (AnthonyLatsis) [reply lib/ASTGen/.swift-format:3]: > If we were to symlink swift-syntax’s config file to swift, it means that you can’t work on swift-syntax as a standalone package anymore.

I meant the symlink to be in the swift repository.

> I think the best thing to do, is to have a separate config file in each repo and add a CI check that all the config files are the same.

Alternatively, maybe we could create a special repo to host this kind of stuff and add it as a git submodule to other repos. Then each repo can replace their own config files with symlinks to the config files in that submodule.

**Pull request merged**